src/realm/object-store/sync/sync_session.cpp

```diff
@@ -468,15 +468,37 @@ void SyncSession::download_fresh_realm(sync::ProtocolErrorInfo::Action server_re
     }
 
     fresh_sync_session->assert_mutex_unlocked();
-    if (auto local_subs_store = m_flx_subscription_store) {
-        sync::SubscriptionSet active = local_subs_store->get_active();
-        auto fresh_sub_store = fresh_sync_session->get_flx_subscription_store();
-        REALM_ASSERT(fresh_sub_store);
-        auto fresh_mut_sub = fresh_sub_store->get_latest().make_mutable_copy();
-        fresh_mut_sub.import(active);
-        std::move(fresh_mut_sub)
-            .commit()
-            .get_state_change_notification(sync::SubscriptionSet::State::Complete)
+    // The fresh realm uses flexible sync.
+    if (auto fresh_sub_store = fresh_sync_session->get_flx_subscription_store()) {
+        auto fresh_sub = fresh_sub_store->get_latest();
+        // The local realm uses flexible sync as well so copy the active subscription set to the fresh realm.
+        if (auto local_subs_store = m_flx_subscription_store) {
```

Comment:
do we want to call get_flx_subscription_store() to access this?

Reply:
it's not needed because `m_state_mutex` is locked at line 449